ompi/communicator/ft/comm_ft.c

```diff
@@ -614,17 +616,23 @@ static int ompi_comm_ishrink_check_setrank(ompi_comm_request_t *request) {
 
     /* cleanup temporary groups */
     OBJ_RELEASE(context->alive_group);
-    context->alive_group = NULL;
     if( NULL != context->alive_rgroup ) {
         OBJ_RELEASE(context->alive_rgroup);
     }
-    context->alive_rgroup = NULL;
 
     /* check errors in prior step */
     if( NULL == *context->newcomm ) {
-        rc = MPI_ERR_INTERN;
         ompi_comm_request_return(request);
+        rc = MPI_ERR_INTERN;
```

Comment:
these two cases are the same, you can't have a successful rc with a NULL newcomm.

Reply:
you're right, in this function `newcomm` cannot be NULL. Looking at this code, I think it has some issues in the case of real faults. Overall, we should not bail out of this entire code without properly updating the peers, or the deadlock is guaranteed, especially if faults already happened and the communicator is already revoked.

For any other errors, aka REVOKE and PROC_FAIL we need to restart the entire loop, which the current code does not seem to be doing. Thus, for now you should remove the check for NULL and keep your `rc` testing code.